hover/effects/background-transitions/_sweep.sass

```diff
@@ -0,0 +1,91 @@
+// duration: duration of the sweep, defaults to $mediumDuration
+
+// Example use:
+// The following will produce a white button with black text. On hover the text will transition from black to white and a ::before element with a black background will sweep up from the bottom to the top. The sweep will last the length of time set with the variable $mediumDuration
+//     .button
+//         @include sweep-vertical(top, black, white)
+//         background: white
+//         color: black
+
+// The following will produce a white button with black text and a 5px border-radius. On hover the text will transition from black to white. An ::after element with a black background will sweep across from left to right. The ::after will also have a 5px border-radius. The sweep will last 0.2s
+//     .button
+//         @include sweep-horizontal(right, black, white, 5px, after, 0.2s)
+//         border-radius: 5px
+//         background: white
+//         color: black
+
+// Styles shared by both sweep mixins
+// THIS MIXIN SHOULD NOT BE USED ON ITS OWN
+@mixin sweep-basics($pseudo, $duration)
+    @include hacks()
+    position: relative
+    @include prefixed(transition-property, color)
+    @include prefixed(transition-duration, #{$duration})
+    &::#{$pseudo}
+        @include prefixed(transition-property, transform)
+        @include prefixed(transition-duration, #{$duration})
+        @include prefixed(transition-timing-function, ease-out)
+        content: ''
+        position: absolute
+        z-index: -1
+        top: 0
+        left: 0
+        right: 0
+        bottom: 0
+
+// Sweep Horizontal - to left or right
+@mixin sweep-horizontal($direction, $hover-bg, $hover-color, $border-radius: 0, $pseudo: before, $duration: $mediumDuration)
+    @include sweep-basics($pseudo, $duration)
+    &::#{$pseudo}
+        @include prefixed(transform, scaleX(0))
+        background: $hover-bg
+        border-radius: $border-radius
+
+        @if $direction == right
+            @include prefixed(transform-origin, 0 50%)
+        @else if $direction == left
+            @include prefixed(transform-origin, 100% 50%)
+
+    &:hover,
+    &:focus,
+    &:active
+        color: $hover-color
+        &::#{$pseudo}
+            @include prefixed(transform, scaleX(1))
+
+// Sweep Vertical - to top or bottom
+@mixin sweep-vertical($direction, $hover-bg, $hover-color, $border-radius: 0, $pseudo: before, $duration: $mediumDuration)
+    @include sweep-basics($pseudo, $duration)
+    &::#{$pseudo}
+        @include prefixed(transform, scaleY(0))
+        background: $hover-bg
+        border-radius: $border-radius
+
+        @if $direction == top
+            @include prefixed(transform-origin, 50% 100%)
+        @else if $direction == bottom
+            @include prefixed(transform-origin, 50% 0)
+
+    &:hover,
+    &:focus,
+    &:active
+        color: $hover-color
+        &::#{$pseudo}
+            @include prefixed(transform, scaleY(1))
```

Comment:
yeah, I was going to at first but there wasn't really a clean way to change the scale, it would just be more if/else. But you're right it would be nicer...